### PR TITLE
Content Draft: Homepage, Getting Started, Concepts

### PR DESCRIPTION
**Pull request by AlannaBurke**
Updated to include information from blog post :

(https://docs.google.com/document/d/1tKPpLZ2ngCT7b_Mwl6oX77alw5Bc_h1m5pnTs1LDdho/edit?tab=t.ysj8o61oflwf) - maybe link this on the readme or in the docs when it's published?

and https://github.com/meta-pytorch/forge/pull/300

### CODE REVIEW

File: docs/source/concepts.md [hunk 1]
```diff
@@ -1,4 +1,516 @@
...
+
+**Actor Mesh**
+: An array of actors, each running inside a separate process
+
+Like array programming in NumPy or PyTorch, meshes make it simple to dispatch operations efficiently across large systems. You can slice meshes, broadcast operations, and operate on entire meshes with simple APIs.
+
+```python
+from monarch.actor import Actor, this_host
+
+# Create a process mesh with 8 GPUs
```
Comment (casteryh) [inline docs/source/concepts.md]: this is potentially misleading. since `gpus` is merely the name of the axis, naming it `gpus` doesn't actually assign the mesh 8 gpus.

Comment (allenwang28) [reply docs/source/concepts.md]: @AlannaBurke  I think we can change all "GPUs" here to "procs" to reduce confusion

File: docs/source/index.md [hunk 2]
```diff
@@ -1,25 +1,234 @@
...
+Before starting significant work, signal your intention in the issue tracker to coordinate with maintainers.
+```
+
+## Documentation Contents
 
 ```{toctree}
-:maxdepth: 1
-:caption: Contents:
+:maxdepth: 2
+:caption: Documentation
```
Comment (svekars) [inline docs/source/index.md:240]: Thinking we can remove the caption completely.
Also, since the page is already long and extensive, could consider adding :hidden: - but I don't feel strongly about that

File: docs/source/index.md [hunk 3]
```diff
@@ -1,25 +1,234 @@
...
+:::{grid-item-card} **TorchTitan**
+:link: https://github.com/pytorch/torchtitan
+
+Meta's production-grade LLM training platform with FSDP, pipeline parallelism, and tensor parallelism.
+
+**Why it matters:** Battle-tested training infrastructure proven at scale
+:::
+
+:::{grid-item-card} **TorchStore**
+
```
Comment (svekars) [inline docs/source/index.md:55]: :link: https://github.com/meta-pytorch/torchstore

File: docs/source/index.md [hunk 4]
```diff
@@ -1,25 +1,234 @@
...
+- **GitHub**: [meta-pytorch/forge](https://github.com/meta-pytorch/forge)
+- **Issues**: [Report bugs and request features](https://github.com/meta-pytorch/forge/issues)
+- **Contributing**: [CONTRIBUTING.md](https://github.com/meta-pytorch/forge/blob/main/CONTRIBUTING.md)
+- **Code of Conduct**: [CODE_OF_CONDUCT.md](https://github.com/meta-pytorch/forge/blob/main/CODE_OF_CONDUCT.md)
+
+```{tip}
+Before starting significant work, signal your intention in the issue tracker to coordinate with maintainers.
+```
+
+## Documentation Contents
```
Comment (svekars) [inline docs/source/index.md]: would remove this 
```suggestion
```

File: docs/source/index.md [hunk 5]
```diff
@@ -29,8 +238,12 @@ api
 faq
 ```
 
-## Indices and tables
+## Indices
+
+* {ref}`genindex` - Index of all documented objects
+* {ref}`modindex` - Python module index
+* {ref}`search` - Search the documentation
```
Comment (svekars) [inline docs/source/index.md]: let's remove the link to search page:
```suggestion
```

File: docs/source/getting_started.md [hunk 6]
```diff
@@ -1,9 +1,490 @@
-# Get Started
+# Getting Started
 
-Welcome to TorchForge! This guide will help you get up and running with TorchForge, a PyTorch-native platform specifically designed for post-training generative AI models.
+Welcome to TorchForge! This guide will walk you through installing TorchForge, understanding its dependencies, verifying your setup, and running your first training job.
```
Comment (svekars) [inline docs/source/getting_started.md]:
```suggestion
This guide will walk you through installing TorchForge, understanding its dependencies, verifying your setup, and running your first training job.
```

File: docs/source/getting_started.md [hunk 7]
```diff
@@ -1,9 +1,490 @@
...
+   - After installing, authenticate with: `gh auth login`
+   - You can use either HTTPS or SSH as the authentication protocol
+
+3. **Git**: For cloning the repository
+   - Usually pre-installed on Linux systems
+   - Verify with: `git --version`
+
+## Understanding TorchForge's Dependencies
+
+TorchForge is built on a carefully curated stack of components, each solving specific challenges in distributed RL. Understanding these dependencies helps you troubleshoot issues and customize your setup.
```
Comment (svekars) [inline docs/source/getting_started.md]: I think we can remove this as it doesn't add any value

File: docs/source/getting_started.md [hunk 8]
```diff
@@ -1,9 +1,490 @@
...
+|-----------|-------------|-------|
+| **Operating System** | Linux (Fedora/Ubuntu/Debian) | MacOS and Windows not currently supported |
+| **Python** | 3.10 or higher | Python 3.11 recommended |
+| **GPU** | NVIDIA with CUDA support | AMD GPUs not currently supported |
+| **CUDA** | 12.8 or higher | Required for GPU training |
+| **Minimum GPUs** | 2 for SFT, 3 for GRPO | More GPUs enable larger models |
+| **RAM** | 32GB+ recommended | Depends on model size |
+| **Disk Space** | 50GB+ free | For models, datasets, and checkpoints |
+
+### Required Tools
```
Comment (svekars) [inline docs/source/getting_started.md]: Would call it Prerequisites
```suggestion
### Prerequisites

Before installing TorchForge, ensure you have the following prerequisites:
```

File: docs/source/getting_started.md [hunk 9]
```diff
@@ -1,9 +1,490 @@
...
+
+1. **Conda or Miniconda**: For environment management
```
Comment (svekars) [inline docs/source/getting_started.md]: Since there is no order in acquiring these dependencies, it should be a bulleted list, right?

File: docs/source/getting_started.md [hunk 10]
```diff
@@ -1,9 +1,490 @@
...
+
+3. **Git**: For cloning the repository
+   - Usually pre-installed on Linux systems
+   - Verify with: `git --version`
+
+## Understanding TorchForge's Dependencies
+
+TorchForge is built on a carefully curated stack of components, each solving specific challenges in distributed RL. Understanding these dependencies helps you troubleshoot issues and customize your setup.
+
+### Monarch: The Distributed Foundation
```
Comment (svekars) [inline docs/source/getting_started.md]:
```suggestion
### Monarch
```

File: docs/source/getting_started.md [hunk 11]
```diff
@@ -1,9 +1,490 @@
...
+- **RDMA Support**: Direct GPU-to-GPU memory transfers for efficient data movement
+
+**What it solves:** Traditional SPMD (Single Program, Multiple Data) approaches require complex coordination logic in your code. Monarch abstracts this away, letting you write RL algorithms naturally while it handles distributed complexity.
+
+**Technical details:** Monarch is implemented with a Python frontend and a Rust backend for performance and robustness. It provides:
+- Scalable messaging with multicast trees
+- Multipart messaging for zero-copy data transfers
+- Integration with PyTorch's distributed primitives
+
+### vLLM: High-Performance Inference
```
Comment (svekars) [inline docs/source/getting_started.md]:
```suggestion
### vLLM
```

File: docs/source/getting_started.md [hunk 12]
```diff
@@ -1,9 +1,490 @@
...
+- **PagedAttention**: Memory-efficient attention mechanism that reduces memory fragmentation
+- **Continuous Batching**: Dynamic batching that maximizes GPU utilization
+- **High Throughput**: Handles generation for multiple concurrent rollouts efficiently
+- **Production-Ready**: Battle-tested at scale with proven performance
+
+**What it solves:** In RL for LLMs, policy generation is often the bottleneck. Autoregressive generation is costly, and blocking training on it kills throughput. vLLM enables fast, efficient inference that doesn't bottleneck your training loop.
+
+**Technical details:** vLLM version 0.10.0+ is required. TorchForge integrates directly with vLLM's engine, giving you access to customize generation strategies, memory management, and inference logic.
+
+### TorchTitan: Production Training Infrastructure
```
Comment (svekars) [inline docs/source/getting_started.md]:
```suggestion
### TorchTitan
```

File: docs/source/getting_started.md [hunk 13]
```diff
@@ -1,9 +1,490 @@
...
+- **FSDP (Fully Sharded Data Parallel)**: Shard parameters, gradients, and optimizer states across GPUs
+- **Pipeline Parallelism**: Split model layers across devices with efficient micro-batching
+- **Tensor Parallelism**: Split individual tensors across devices for very large models
+- **Proven at Scale**: Used to train Llama models on thousands of GPUs
+
+**What it solves:** Modern models are too large to fit on single GPUs. TorchTitan provides the sophisticated sharding and parallelism strategies needed to train them efficiently, with optimizations battle-tested in production.
+
+**Technical details:** TorchForge integrates with TorchTitan for training step logic and sharding strategies, enabling experimentation without framework constraints.
+
+### TorchStore: Weight Synchronization
```
Comment (svekars) [inline docs/source/getting_started.md]:
```suggestion
### TorchStore
```

File: docs/source/getting_started.md [hunk 14]
```diff
@@ -1,9 +1,490 @@
...
+- **Automatic Resharding**: Handles complex weight transfer between different sharding strategies
+- **DTensor Support**: Native support for distributed tensors
+- **RDMA Transfers**: High-bandwidth weight movement without synchronous GPU transfers
+- **Asynchronous Updates**: Training and inference can read/write weights independently
+
+**What it solves:** In async RL, new policy weights must propagate to all inference replicas. For a 70B parameter model across 16 replicas, this means moving hundreds of gigabytes. TorchStore makes this efficient, handling resharding automatically and using RDMA for fast transfers.
+
+**Technical details:** TorchStore provides a simple key-value interface while optimizing data movement behind the scenes, staying distributed across the cluster until requested.
+
+### PyTorch Nightly: Cutting-Edge Features
```
Comment (svekars) [inline docs/source/getting_started.md]:
```suggestion
### PyTorch Nightly
```

File: docs/source/getting_started.md [hunk 15]
```diff
@@ -1,9 +1,490 @@
...
+- **Advanced Memory Management**: Latest FSDP and memory optimization features
+- **Bug Fixes**: Continuous improvements to distributed training primitives
+
+**Installation note:** The installation script provides pre-built wheels with PyTorch nightly already included.
+
+## Installation
+
+TorchForge offers two installation methods. Choose the one that fits your setup:
+
+### Method 1: Basic Installation (Recommended)
```
Comment (svekars) [inline docs/source/getting_started.md]:
```suggestion
### Basic Installation (Recommended)
```

File: docs/source/getting_started.md [hunk 16]
```diff
@@ -1,9 +1,490 @@
...
+
+**Step 1: Clone the Repository**
+
+```bash
+git clone https://github.com/meta-pytorch/forge.git
+cd forge
+```
+
+**Step 2: Create Conda Environment**
+
```
Comment (svekars) [inline docs/source/getting_started.md]: would just make it a simple 1,2,3 list

File: docs/source/getting_started.md [hunk 17]
```diff
@@ -1,9 +1,490 @@
...
+Test that TorchForge is properly installed:
+
+```bash
+python -c "import forge; print(f'TorchForge version: {forge.__version__}')"
+python -c "import monarch; print('Monarch: OK')"
+python -c "import vllm; print(f'vLLM version: {vllm.__version__}')"
+```
+
+### Method 2: Meta Internal Installation (Alternative)
```
Comment (svekars) [inline docs/source/getting_started.md]: Do we need to have internal instructions here? My guess not @allenwang28

File: docs/source/getting_started.md [hunk 18]
```diff
@@ -1,9 +1,490 @@
...
+
+### Check CUDA Version
+
+```bash
+python -c "import torch; print(f'CUDA version: {torch.version.cuda}')"
+```
+
+Expected output: `CUDA version: 12.8` (or higher)
+
+### Check All Dependencies
```
Comment (svekars) [inline docs/source/getting_started.md]: make it a simple list, not headings

File: docs/source/getting_started.md [hunk 19]
```diff
@@ -1,9 +1,490 @@
...
+
+Now that you have TorchForge installed and verified:
+
+1. **Learn the Concepts**: Read {doc}`concepts` to understand TorchForge's architecture, including Monarch, Services, and TorchStore
+2. **Explore Examples**: Check the `apps/` directory for more training examples
+3. **Customize Training**: See {doc}`usage` for configuration patterns
+4. **Read Tutorials**: Follow {doc}`tutorials` for step-by-step guides
+5. **API Documentation**: Explore {doc}`api` for detailed API reference
+
+## Getting Help
```
Comment (svekars) [inline docs/source/getting_started.md:263]: I think this could go to the Home page maybe? Or a separate page? Could be another horizontal bar item.

File: docs/source/getting_started.md [hunk 20]
```diff
@@ -1,9 +1,490 @@
...
+2. **GitHub CLI (gh)**: Required for downloading pre-packaged dependencies
+   - Install instructions: [github.com/cli/cli#installation](https://github.com/cli/cli#installation)
+   - After installing, authenticate with: `gh auth login`
+   - You can use either HTTPS or SSH as the authentication protocol
+
+3. **Git**: For cloning the repository
+   - Usually pre-installed on Linux systems
+   - Verify with: `git --version`
+
+## Understanding TorchForge's Dependencies
```
Comment (svekars) [inline docs/source/getting_started.md]: I feel like this whole section might be too much for the getting started. Maybe we move it to the concepts? 
We could just have a short list of 4 main dependencies and add for more information, see <link>

File: docs/source/getting_started.md [hunk 21]
```diff
@@ -1,9 +1,490 @@
-# Get Started
+# Getting Started
```
Comment (svekars) [inline docs/source/getting_started.md:5]: since this is the only page in the whole Getting Started section, you end up with an empty left navigation. Add orphan true metadata to all pages that are like that (see screenshot)
```suggestion
---
orphan: true
---

# Getting Started
```
Screenshot 2025-10-10 at 10 34 35 AM

File: docs/source/concepts.md [hunk 22]
```diff
@@ -1,4 +1,516 @@
 # Concepts
 
-This guide covers the fundamental concepts and architecture behind TorchForge,
-helping you understand how the system works and how to effectively use its components.
+This guide covers the fundamental concepts and architecture behind TorchForge, helping you understand how the system works and how to effectively use its components.
```
Comment (svekars) [inline docs/source/concepts.md]: for the concepts section I feel like there is a lot of text but hard to understand what those things are and how they are connected. Maybe we move ## Understanding TorchForge's Dependencies here and then also add what's in the tutorials PR?

**Comment by AlannaBurke** [comment]
@svekars i think i incorprated all of your changes - i also split the concepts page into smaller pages since it was so long. Have another look?

File: docs/source/getting_started.md [hunk 23]
```diff
@@ -1,9 +1,316 @@
...
+print(f'GPUs: {torch.cuda.device_count()}')
+"
+```
+
+Include this output in your bug reports!
+
+## Additional Resources
+
+- **GitHub Repository**: [github.com/meta-pytorch/forge](https://github.com/meta-pytorch/forge)
+- **Example Notebooks**: Check `demo.ipynb` in the repository root
```
Comment (svekars) [inline docs/source/getting_started.md]: suggesting to remove those. We link to the repo in the nav bar and I don't think demo.ipynb is relevant.
```suggestion
```

File: docs/source/faq.md [hunk 24]
```diff
@@ -1,3 +1,57 @@
 # FAQ
 
-This FAQ covers common questions and issues encountered when using TorchForge.
+## Common Installation Issues
```
Comment (svekars) [inline docs/source/faq.md]: I think we agreed not to have FAQ section for now, so we can delete this -

Comment (AlannaBurke) [reply docs/source/faq.md]: Do you think these troubleshooting faqs are valuable info? i was just trying to reduce the length of the other pages - should i put these back with the installation, or just leave it for now and we can add it back later if we want?

Comment (svekars) [reply docs/source/faq.md]: I don't have a strong opinion on this - feel free to leave as is.

**Comment by svekars** [comment]
@allenwang28 @pbontrager - can you please review this

File: docs/source/architecture.md [hunk 25]
```diff
@@ -0,0 +1,256 @@
...
+class PolicyActor(Actor):
+    @endpoint
+    def generate(self, prompt):
+        return self.model.generate(prompt)
+
+# Spawn actors across the mesh
+actors = procs.spawn("policy", PolicyActor)
+
+# Call methods on the entire mesh
+results = actors.generate.call_all("Hello world")
```
Comment (allenwang28) [inline docs/source/architecture.md]: this should just be `call()`

File: docs/source/getting_started.md [hunk 26]
```diff
@@ -1,9 +1,314 @@
...
+
+**What's Happening:**
+- GPU 0: Policy model (being trained, powered by TorchTitan)
+- GPU 1: Reference model (frozen baseline)
+- GPU 2: Reward model (scoring outputs, powered by vLLM)
+- **Monarch** orchestrates all three components
+- **TorchStore** handles weight synchronization from training to inference
+
+**Expected Output:**
+```
```
Comment (svekars) [inline docs/source/getting_started.md]:
```suggestion
```bash
```
